### PR TITLE
Pro Plan: add included features to plan definition

### PR DESCRIPTION
**Pull request by michaeldcain**
Some feature gating (e.g. "Google My Business") is handled via `planHasFeatures`. This PR adds the `getIncludedFeatures` object from the Business plan.

See: c/1KciuyFW-tr

**To test:**
- on a site with a Pro plan
- visit `/plans/my-plan`
- scroll down and click "Connect to Google My Business"
- verify that the Google My Business screen is shown

### CODE REVIEW

**Review by DavidRothstein — APPROVED** [review]
I confirmed the fix works.  After the similar fix in https://github.com/Automattic/wp-calypso/pull/62376 I was wondering if we might run into more of these... I guess the answer is yes :smile: 

This is fine as is, but since you already decided to add a bunch of them at once, I just want to point out that that part isn't entirely complete:

>This PR adds the getIncludedFeatures object from the Business plan.

But what's really needed is to add **all the features** from Business, not just `getIncludedFeatures` (except ones where we specifically want the Pro plan to be different).  It probably doesn't matter where they go, so dumping the remaining ones in `getIncludedFeatures` would be fine, regardless of where they were for Business.

As an example, I was able to find another bug from `FEATURE_REPUBLICIZE` being missing in the Pro plan (see below).  We don't have to fix them all at once, of course, but I just wanted to mention it here because it might be easier to do all at once.

When going to the Posts page and using the "Share" option to share a post to a connected account, on Business I get this:

business-plan

But on Pro I get this:

pro-plan

**Comment by michaeldcain** [comment]
@DavidRothstein good catch. In 7b108bd I went ahead and dumped missing features (that I think we intend to include with Pro) in `getIncludedFeatures()` as a precautionary measure. Some don't line up to actual features, but rather to marketing feature display, but it should be a decent temporary fix. We should ultimately port this to the feature work that the Limited Atomic pod has been working on.

File: packages/calypso-products/src/plans-list.tsx [hunk 1]
```diff
@@ -1591,6 +1591,7 @@ PLANS_LIST[ PLAN_WPCOM_FLEXIBLE ] = {
 };
 
 PLANS_LIST[ PLAN_WPCOM_PRO ] = {
+	...getDotcomPlanDetails(),
```
Comment (DavidRothstein) [inline packages/calypso-products/src/plans-list.tsx:1594]: Does this do anything?  It adds `getAnnualPlansOnlyFeatures()` but those don't seem to get checked in `hasFeature()` calls.  (Also, the Pro plan only has annual plans anyway.)

But as far as I can tell, it doesn't cause any harm either.

File: packages/calypso-products/src/plans-list.tsx [hunk 2]
```diff
@@ -1622,4 +1623,26 @@ PLANS_LIST[ PLAN_WPCOM_PRO ] = {
...
+		FEATURE_GOOGLE_ANALYTICS,
+		FEATURE_GOOGLE_MY_BUSINESS,
+		FEATURE_HOSTING,
+		FEATURE_MEMBERSHIPS,
+		FEATURE_NO_BRANDING,
+		FEATURE_REPUBLICIZE,
+		FEATURE_PREMIUM_CONTENT_BLOCK,
+		FEATURE_SEO_PREVIEW_TOOLS,
+		FEATURE_SFTP_DATABASE,
+		FEATURE_SIMPLE_PAYMENTS,
```
Comment (DavidRothstein) [inline packages/calypso-products/src/plans-list.tsx]: This one is already listed above so doesn't need to be added here (but there's presumably no harm either).

**Review by DavidRothstein — COMMENTED** [review]
Looks good to me.  I checked the list of features added and they're all part of the Business plan also, and all seem to make sense for the Pro plan too.

I think there are a couple still listed on the Business plan that didn't make it in here, but as far as I can tell none of those missing ones are actually used in `hasFeature()` calls around the codebase anyway... And as you said, this is a legacy system that is hopefully going to be replaced soon anyway.

I also double-checked that the latest version of the code here now also solves the Publicize bug I mentioned.

**Comment by DavidRothstein** [comment]
>I think there are a couple still listed on the Business plan that didn't make it in here, but as far as I can tell none of those missing ones are actually used in `hasFeature()` calls around the codebase anyway...

I didn't search well enough, apparently, since there's also a `planHasFeature()` method that does basically the same thing as `hasFeature()`.  As a result, it looks like https://github.com/Automattic/wp-calypso/pull/62529 added one more feature to the list that's needed here.  But after that, I'm pretty sure this is good :smile:

**Pull request merged**